pkg/stanza/operator/transformer/recombine/recombine.go

```diff
@@ -267,73 +277,84 @@ func (r *Transformer) matchIndicatesLast() bool {
 
 // addToBatch adds the current entry to the current batch of entries that will be combined
 func (r *Transformer) addToBatch(_ context.Context, e *entry.Entry, source string) {
-	if _, ok := r.batchMap[source]; !ok {
-		r.batchMap[source] = []*entry.Entry{e}
+	if batch, ok := r.batchMap[source]; !ok {
+		r.batchMap[source] = &sourceBatch{
+			entries:    []*entry.Entry{e},
+			recombined: strings.Builder{},
+		}
 		if len(r.batchMap) >= r.maxSources {
 			r.Error("Batched source exceeds max source size. Flushing all batched logs. Consider increasing max_sources parameter")
 			r.flushUncombined(context.Background())
+			return
 		}
+	} else {
+		batch.entries = append(batch.entries, e)
+	}
+	batch := r.batchMap[source]
+	// Combine the combineField of each entry in the batch,
+	// separated by newlines
+	var s string
+	err := e.Read(r.combineField, &s)
+	if err != nil {
+		r.Errorf("entry does not contain the combine_field")
 		return
 	}
+	if batch.recombined.Len() > 0 {
+		batch.recombined.WriteString(r.combineWith)
+	}
+	batch.recombined.WriteString(s)
 
-	r.batchMap[source] = append(r.batchMap[source], e)
-	if len(r.batchMap[source]) >= r.maxBatchSize {
+	if r.maxLogSize > 0 {
+		if int64(batch.recombined.Len()) > r.maxLogSize {
+			if err := r.flushSource(source); err != nil {
+				r.Errorf("there was error flushing combined logs %s", err)
+			}
+			return
+		}
+	}
+
+	if len(batch.entries) >= r.maxBatchSize {
```

Comment:
I think we should combine at least the two max log size conditions, but probably also the max batch size condition too, since all ends the same way if either condition is met.

```suggestion
	if (r.maxLogSize > 0 && int64(batch.recombined.Len()) > r.maxLogSize) || len(batch.entries) >= r.maxBatchSize {
```

Reply:
I separated them before because I thought it would make the code clearer. But I agree to combine because there are only two conditions so far.